utils.py

```diff
@@ -325,3 +362,23 @@ def version():
 
 def valid_email(s):
     return bool (re.match ('^(([a-zA-Z0-9_+\.\-]+)@([\da-zA-Z\.\-]+)\.([a-zA-Z\.]{2,6})\s*)$', s))
+
+
+@contextlib.contextmanager
+def atomic_write_file(filename, mode='wb'):
+    """Write to a filename atomically.
+
+    First write to a unique tempfile, then rename the tempfile into
+    place. Use as a context manager:
+
+        with atomic_write_file('file.txt') as f:
+            f.write('hello')
+
+    THIS WON'T WORK ON WINDOWS -- atomic file renames don't overwrite
```

Comment:
Would this limitation break only production environments on Windows? Or also dev environments? We currently have a few contributors working from Windows.

Reply:
This will only be used in production, and all of our production is Linux... but good point. I was being lazy. I will add safeguards just to be sure.

Reply:
Thank you!